packages/navie/src/commands/observe-command.ts

```diff
@@ -0,0 +1,143 @@
+          content: userRequest,
+        },
+      ],
+      RelevantTest,
+      { model: ModelType.Mini }
+    );
+
+    return response?.relevantTest;
+  }
+
+  async *execute({ question: userRequest, userOptions }: CommandRequest): AsyncIterable<string> {
+    const vectorTerms = await this.vectorTermsService.suggestTerms(userRequest);
+    const tokenLimit = userOptions.numberValue('tokenlimit') || this.options.tokenLimit;
+    const testSnippets = await this.getTestSnippets(vectorTerms, tokenLimit);
+    const mostRelevantTest = await this.getMostRelevantTest(userRequest, testSnippets);
+    if (!mostRelevantTest) {
+      yield 'Sorry, I could not find any relevant tests to record.';
+      return;
+    }
+
+    if (mostRelevantTest.language === 'other') {
+      yield `I found a relevant test at \`${mostRelevantTest.path}\`, but I'm unable to help you record it at this time. This language does not appear to be supported.`;
+      return;
+    }
+
+    const helpDocs = await this.lookupContextService.lookupHelp(
+      [mostRelevantTest.language],
+      ['record', 'agent', 'tests', mostRelevantTest.framework].filter(Boolean) as string[],
+      tokenLimit
+    );
+
+    const messages: ReadonlyArray<Message> = [
+      {
+        role: 'system',
+        content: `You are Navie, an AI assistant created by AppMap. Your job is to help the user identify a relevant test case and record AppMap trace data for that test case.`,
+      },
+      {
+        role: 'user',
+        content: `Identify the most relevant test case from the following string:
+\`\`\`
+${userRequest}
+\`\`\``,
+      },
+      {
+        role: 'assistant',
+        content: `Based on the request, the most relevant test case is:
+${mostRelevantTest.name ? `**Name:** \`${mostRelevantTest.name}\`` : ''}
+${mostRelevantTest.framework ? `**Framework:** \`${mostRelevantTest.framework}\`` : ''}
+${mostRelevantTest.language ? `**Language:** \`${mostRelevantTest.language}\`` : ''}
+**Path:** \`${mostRelevantTest.path}\`
+
+Documentation which may be relevant to recording this particular test case is provided below.
+<documentation>
+${helpDocs
+  .map(
+    (doc) =>
+      `<document file="${doc.filePath}" from="${doc.from}" to="${doc.to}">${doc.content}</document>`
+  )
+  .join('\n')}
+</documentation>`,
+      },
+      {
+        role: 'user',
+        content: `Restate the information you've provided to me, in standalone format, as a step by step guide outlining the steps required to record the single test case that you've identified.
+If possible, include the terminal command needed to run the test. Avoid specifying file ranges in command line arguments. Only specify test patterns that are guaranteed to match based on previous context.
+Only provide instructions related to AppMap if they appear previously in the documentation. Ignore instructions that are phrased with tentative language such as "if you don't see this working". Assume everything works as expected.
```

Comment:
If you require the AI to respond in a structured way - for example, give it an option to provide a terminal command, an option to specify an install command, or  code modification to activate the agent, then you may have less of a problem with free text instructions that include steps you don’t want it to include. 

And in a structured format, we would also have a better chance of being able to run the suggested command automatically.